### ISSUE TITLE
[Feature Request] being able to switch between sweeper without changing sweeper config

### ISSUE THREAD

**Issue by Harimus** [facebookresearch/hydra#1615]
# 🚀 Feature Request
I'm not sure if this is a feature request, or I'm just doing it the wrong way and there's a correct way to do this.
So I currently use Hydra and AX sweeper for an RL experiment repo, and in the base `config.yaml` file I have the defaults as
```
defaults:
  - algorithm: something 
  - hydra/sweeper: ax
  - hyperparam_opt: file_with_ax_config
```
with `hyperparam_opt` being a subfolder containing multiple yaml file with different sweeper settings for AX. They only contain text in the format
```
# @package _global_
hydra:
  sweeper:
    ax_config:
      max_trials: 30
      .......etc
```
with global scope. 
Now there's two thing I want to do: 
* Run normal hydra sweeper (reason: seed run does not need AX).  
* Run normal hydra on a environment without AX installed (reason: AX requires a specific pyTorch version, and this have broke my coworkers environment when he unknowingly running things)

This can be done by setting at runtime (or change config) of `hydra/sweeper`:  to `base` **and** add an empty file in `hyperparam_opt` subfolder. Something like an `empty.yaml` file containing:
```
# @package _global_
placeholder_param: False
```

 If I only change `hydra/sweeper` to `basic` and **not** the `hyperparam_opt` to `empty` the code errors out with the message

```
Error merging hyperparam_opt=file_with_ax_config
Key 'ax_config' not in 'BasicSweeperConf'
	full_key: hydra.sweeper.ax_config
	reference_type=Optional[BasicSweeperConf]
	object_type=BasicSweeperConf
```

The behavior I'd like to have as an feature is that the basic sweeper of hydra just *ignores* the nested `ax_config` parameters, instead of crashing. 

so instead of writing (at runtime)
`python3 main.py hydra/sweeper=basic hyperparam_opt=empty seed=1,2,3,4,5`, 
`python3 main.py hydra/sweeper=basic seed=1,2,3,4,5` would suffice.

This will also remove the `empty.yaml` file from the subfolder (which otherwise probably have to include an comment with an explanation why it exist). I understand that there's multiple sweeper versions (and not just basic and ax) but couldn't the basic behavior be to *ignore* anything outside the specified parameters one can set for the sweeper specified in the documentation?

## Additional context
A way to circumvent the additional typing in the terminal to run this is to just add an additional `no_ax_config.yaml` which have the additional `hydra/sweeper: ax` and  `hyperparam_opt: file_with_ax_config` removed from `default`, but otherwise identical to `config.yaml`. 

So if this request goes against API design ideas I could just work with two base config files.

**Comment by jieru-hu** [comment]
Could you check out this doc for [configuring experiments](https://hydra.cc/docs/next/patterns/configuring_experiments), this sounds like a good use case for that.

Instead of having your defaults list in config file, you can have them in a config group (for example `experiment` like the doc) and from the command line you can override `experiment=use_ax` instead of having to do two overrides.

**Comment by jieru-hu** [comment]
thanks @omry 
@Harimus  Hydra 1.1 is very close to a RC and has support for recursive defaults (which better support your use case here). Please give it a try if you can :) 
to install, run `pip install hydra-core --pre --upgrade`

**Comment by omry** [comment]
The title of your issue is "being able to switch between sweeper without changing sweeper config".
This can be interpreted as one of:
1. I want to be able to switch between Sweeper configs easily
2. I want all Sweepers to have compatible configs, so the same config would work for all Sweepers I use.

1 is supported:
Be sure that you are using Hydra 1.1.
Check the doc for [configuring plugins](https://hydra.cc/docs/next/patterns/configuring_plugins).
As @jieru-hu pointed out, recursive defaults support is what makes the difference here. This is the key new feature of Hydra 1.1.

2 is not supported:
While we intend to eventually standardize some aspects of hyper parameter optimization sweepers it will take some time and in addition there will always  be differences.

**Comment by Harimus** [comment]
Hello! sorry for the late reply, I've checked out the doc linked by @jieru-hu and it indeed would work to reduce the command line argument (and make it more intuitive) by moving over the `hydra/sweeper: ax` to that of experiments.  Thank you very much for the suggestion?

While at it, can I also ask if there's a way to specify the parameters to be sweeped for the basic sweeper, but in config.yaml files rather than on command line? The tutorial only showed how to specify it on runtime while[ hydra/ax tutorial page](https://hydra.cc/docs/plugins/ax_sweeper/) gives a config file example

**Comment by jieru-hu** [comment]
@Harimus awesome!

We already have a issue tracking configuring via Yaml for basic sweeper, please subscribe to this issue https://github.com/facebookresearch/hydra/issues/1376


Thanks!

**Comment by Harimus** [comment]
@jieru-hu Thanks! will do that!

Also wanted to thank you guys for hydra and the work you put in to it! Amazing thing and have saved me huge amount of headache :D you guys rock! :+1:

**Comment by jieru-hu** [comment]
Thank you for the kind words! 🙂  Glad to hear that!

I'm closing this issue now. Pls feel free to follow up if needed. :)
